### PR TITLE
Add IR size after certain phases to the JIT log.

### PR DESCRIPTION
**Pull request by pgavlin**
This change adds support for measuring and reporting the number of nodes
in a function after a certain subset of the JIT's phases. This reporting
is enabled by setting `COMPlus_JitMeasureIR` and `COMPlus_JtiTimeLogCsv`
in conjunction with each other.

### CODE REVIEW

**Comment by pgavlin** [comment]
@BruceForstall @AndyAyersMS @dotnet/jit-contrib PTAL

**Comment by pgavlin** [comment]
@BruceForstall @AndyAyersMS @dotnet/jit-contrib PTAL

**Comment by pgavlin** [comment]
> Would also be interesting to know how many dead IR nodes there are.

Can you file an issue and assign it to me? This is easy to calculate if we're willing to maintain a running total of the # of IR nodes created.

**Comment by AndyAyersMS** [comment]
#9569

**Comment by BruceForstall** [comment]
LGTM

You could add a dump of block count, lclvar/tmp count, etc., on a per-phase basis.

You could imagine a more generic UI to avoid pre-determining the set your plan to collect by listing the phases for which you want the info (using the "short names"):

```
set COMPlus_JitMeasureIR=REGALLOC;LSRA
set COMPlus_JitMeasureIR=* // special: * == all phases
set COMPlus_JitMeasureIR=default or 1 or true // special: some pre-determined "interesting" set, like what you've defined
```

File: src/jit/flowgraph.cpp [hunk 1]
```diff
@@ -24256,3 +24256,36 @@ void Compiler::fgTransformFatCalli()
     CheckNoFatPointerCandidatesLeft();
 #endif
 }
+
+//------------------------------------------------------------------------
+// fgMeasureIR: count and return the number of IR nodes in the function.
+//
+unsigned Compiler::fgMeasureIR()
```
Comment (mikedn) [inline src/jit/flowgraph.cpp]: 30 more lines in a 24000 lines file 😁

**Comment by pgavlin** [comment]
> You could add a dump of block count, lclvar/tmp count, etc., on a per-phase basis.
> 
> You could imagine a more generic UI to avoid pre-determining the set your plan to collect by listing the phases for which you want the info (using the "short names"):

These are both good suggestions. I've filed #9578 to track.

**Pull request merged**